st2actions/st2actions/runners/python_action_wrapper.py

```diff
@@ -114,14 +114,27 @@ def __init__(self, pack, file_path, parameters=None, user=None, parent_args=None
     def run(self):
         action = self._get_action_instance()
         output = action.run(**self._parameters)
+        action_status = None
+
+        if type(output) is tuple and len(output) is 2:
+            action_status = output[0]
+            action_result = output[1]
+        else:
+            action_result = output
+
+        action_output = {"result": action_result}
 
         # Print output to stdout so the parent can capture it
         sys.stdout.write(ACTION_OUTPUT_RESULT_DELIMITER)
         print_output = None
         try:
-            print_output = json.dumps(output)
+            if action_status:
+                action_output['status'] = action_status
+                print_output = json.dumps(action_output)
+            else:
+                print_output = json.dumps(action_output)
         except:
-            print_output = str(output)
+            print_output = str(action_output)
```

Comment:
Hm, do we indeed want to use `action_output` here or would the right thing be to just use `action_result`?